core/core.go

```diff
@@ -471,9 +471,10 @@ func (n *IpfsNode) loadBootstrapPeers() ([]peer.PeerInfo, error) {
 // uses it to instantiate a routing system in offline mode.
 // This is primarily used for offline ipns modifications.
 func (n *IpfsNode) SetupOfflineRouting() error {
-	err := n.LoadPrivateKey()
-	if err != nil {
-		return err
+	if n.PrivateKey == nil {
+		if err := n.LoadPrivateKey(); err != nil {
```

Comment:
maybe instead of having this, we should probably just make `LoadPrivateKey` not fail if called while the key is already loaded